base/complex.jl

```diff
@@ -107,7 +107,7 @@ abs(z::Complex)  = hypot(real(z), imag(z))
 abs2(z::Complex) = real(z)*real(z) + imag(z)*imag(z)
 inv(z::Complex)  = conj(z)/abs2(z)
 inv{T<:Integer}(z::Complex{T}) = inv(float(z))
-sign(z::Complex) = z/abs(z)
+sign(z::Complex) = z == 0? float(zero(z)) : z/abs(z)
```

Comment:
Also, maybe `ifelse(z == 0, float(zero(z)), z/abs(z))` would be faster, since we don't need the short-circuiting behavior of `.. ? .. : ..` here?